GVFS/GVFS/CommandLine/SparseVerb.cs

```diff
@@ -93,6 +93,44 @@ private enum SetDirectoryTimeResult
 
         protected override string VerbName => SparseVerbName;
 
+        internal static bool PathCoveredBySparseFolders(ref int index, string statusOutput, HashSet<string> sparseFolders, out string gitPath)
+        {
+            int endOfPathIndex = statusOutput.IndexOf(StatusPathSeparatorToken, index);
+            gitPath = statusOutput.Substring(index, endOfPathIndex - index);
+            index = endOfPathIndex + 1;

```

Comment:
Just thinking here and wondering if it would make the tests easier and more specific if we pulled this out into something like:
```
private static string GetNextGitPath(ref int index, string statusOutput)
{
    int endOfPathIndex = statusOutput.IndexOf(StatusPathSeparatorToken, index);
    gitPath = statusOutput.Substring(index, endOfPathIndex - index);
    index = endOfPathIndex + 1;
    return gitPath;
}
```
Then this method could just be 
```
internal static bool PathCoveredBySparseFolders(string gitPath, HashSet<string> sparseFolders)
```
`GetPathsNotCoveredBySparseFolders` would need to change to something like
```
string gitPath = GetNextGitPath(ref index, statusOutput);
if (!PathCoveredBySparseFolders(gitPath, sparseFolders))
{
...
```
Then the tests wouldn't need to be testing the parsing of the status output although it would be good to have some coverage there as well.  So I think after writing this I am more on the fence.

Reply:
I like this suggestion, thanks!  I will update `SparseVerb` and add tests for `GetNextGitPath`.